src/inject/dynamic-theme/style-manager.ts

```diff
@@ -186,6 +186,7 @@ export function manageStyle(element: StyleElement, {update, loadingStart, loadin
             // so need to load CSS text and insert it into style element
             try {
                 const fullCSSText = await replaceCSSImports(cssText, cssBasePath);
+                alreadyReplaced.clear();
```

Comment:
Is there a need to clear it here? Suppose 2 `<link>` elements are loading CSS files at the same time. You will clear all for them. Cleaning the cache sounds reasonable for freeing up the memory.

Maybe try restructuring the `replaceCSSImports` like:
```javascript
async function replaceCSSImports(cssText: string, basePath: string, cache: Map<string, string>) {
    // ...
    importedCSS = await replaceCSSImports(importedCSS, getCSSBaseBath(absoluteURL), cache);
    // ...
}
```
Then you will use this function like:
```javascript
const cssImportCache = new Map<string, string>();
const fullCSSText = await replaceCSSImports(cssText, cssBasePath, cssImportCache);
cssImportCache.clear(); // even if you leave it without clear(), Garbage Collector should clean up the memory, because `cssImportCache` variable looses all the references to it when the function exits.
```
Then there will be no need to export the function.

Or even better would be to clear it in `finally` block, that is called when there is an error or no error like:
```javascript
const cssImportCache = new Map<string, string>();
try {
    // ...
} catch(err) {
    // ...
} finally {
    cssImportCache.clear();
}
// or just do `cssImportCache.clear();` here, after `try...catch` without `finally`, should be no difference in this case.
```

Reply:
I've just added a extra variable to pass along the cache. So the garbage collector will remove free the memory of it when the function is done 🤷🏽‍♀️.